### DISCUSSION TITLE
[Feature] Add an image quality score (either for the "x years since" feature and/or as a filter for the user)

### DISCUSSION THREAD

**Discussion by FPMilan** [immich-app/immich#7202] [Feature Request]
### The feature

I like that the popular cloud photo apps manage to show only your best shots.
I was wondering if this functionality can somehow be added to Immich through some ML model like this one: https://github.com/idealo/image-quality-assessment

I can imagine two use cases:
1. Hide the ranking from the user but use it in the backend to select fotos for the "x years since" selection and/or
2. Let the user filter it to find their best fotos manually

Appreciate any thought about this.

(although I did search for this I can not guarantee a similar thing does not exist already or was asked before - if so, sorry)

### Platform

- [X] Server
- [X] Web
- [X] Mobile

**Comment by bo0tzz** [comment]
It would also be nice to use a score like this to (either automatically or manually) reject low-quality images.

**Reply by AngelaDMerkel** [reply]
This might be nicely bundled in a "clean up library" style feature where suspected duplicates or low quality images are surfaced for manual review and possible removal.

**Reply by encodedLife** [reply]
@AngelaDMerkel like here #9822

**Reply by ryleu** [reply]
i've used one of those tinder-like photo deleting apps, and if you ignore the awful privacy policies, they're great!

**Reply by AngelaDMerkel** [reply]
@encodedLife Yes, something like this. Google includes this in Google Photos, but ideally I'd like some kind of internal (local) system. Having the context of my entire library is what matters most too as older photos and accumulated cruft from before phones had more granular control mean I have some truly terrible images; and, too many meme photos.

**Comment by jonvdveen** [comment]
Throwing my support behind this feature request and adding some detail.

I would like to propose the development of image culling features that would help users clean up image clutter and distill their libraries down to the images and memories that matter to them. In my experience, the MVP version of this would include the following key features:

- Enter a special review/culling mode
- Mark a photo as flagged or rejected with one keystroke (or tap)
- Rate a photo with one keystroke (or tap)
- View the flag and star rating from the thumbnail view
- Filter images by flagged, unflagged, rejected, rating, etc.
- Multi-select images and batch apply flags and ratings

Google Photos and Apple Photos are both pretty terrible at this. The more photos and videos that people accumulate, the more cloud storage they're likely to purchase. I think this is a feature that would be broadly beneficial to Immich users.

I'm hoping that this would be a relatively straightforward implementation. I have some ideas for how Immich might build on this in the future, but if the above MVP features were implemented, I would be pleased as punch!

Cheers! 🍻

**Reply by JoshC1994** [reply]
like to add the suggestions made over at #25050

**Comment by headtr1ck** [comment]
Just stumbled upon this tool that looks very promising: https://github.com/prasadabhishek/photo-quality-analyzer
